[Automation] java run fails for io.netty:netty-transport-native-unix-common:4.1.57.Final

## native-image build on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image build
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.54.Final" ./gradlew clean nativeTestCompile -Pcoordinates="io.netty:netty-transport-native-unix-common:4.1.42.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25239750106/job/74013375139
Last 300 lines of the log:
```console
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/8910/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/libs/io.netty.netty-transport-native-unix-common_tests.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.graalvm.buildtools/junit-platform-native/1.0.0/2974f834a2f0a0105f2284fba7ef16dc4d60de56/junit-platform-native-1.0.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/tck-test-support/build/libs/tck-test-support-1.0.0-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.netty/netty-transport-native-unix-common/4.1.54.Final/ac7d48202cf62c8d51abe89f228d338a41cb49cd/netty-transport-native-unix-common-4.1.54.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.netty/netty-transport/4.1.54.Final/ddb9a255819b87b8812932d1d1001f4d9dc7f89e/netty-transport-4.1.54.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.netty/netty-buffer/4.1.54.Final/15c513ef8a2e56e88d3736752ea10eae758f47d8/netty-buffer-4.1.54.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.netty/netty-resolver/4.1.54.Final/7eed781e7531b990f8360e1a5010b53ca73878f5/netty-resolver-4.1.54.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.netty/netty-common/4.1.54.Final/8c330383cda87204ab38e7401ab0f56c0d43b799/netty-common-4.1.54.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=io.netty.netty-transport-native-unix-common_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.netty/netty-transport-native-unix-common/4.1.42.Final,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.netty/netty-common/4.1.80.Final,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.netty/netty-buffer/4.1.80.Final,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.netty/netty-transport/4.1.80.Final,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.netty/netty-resolver/5.0.0.Alpha2 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=d4b90692-7507-fb5f-d9b6-f02a7797c044 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'io.netty.netty-transport-native-unix-common_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.4s @ 0.14GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  []                                                                         (5.7s @ 0.22GB)
    1,790 types,     954 fields, and   3,439 methods found reachable
      987 types,     557 fields, and   3,080 methods registered for reflection
        0 downcalls and 0 upcalls registered for foreign access

Error: parsing had failed in another thread
Caused by: jdk.graal.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field io.netty.util.NetUtil.LOCALHOST4, accessors class io.netty.util.NetUtilSubstitutions$NetUtilLocalhost4Accessor: found no method named set or setLOCALHOST4
	at parsing io.netty.util.NetUtil.<clinit>(NetUtil.java:139)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2802)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:555)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3853)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.iterateBytecodesForBlock(SharedGraphBuilderPhase.java:1269)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3805)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3652)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.build(BytecodeParser.java:1193)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.build(SharedGraphBuilderPhase.java:245)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1076)
	at jdk.graal.compiler/jdk.graal.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:103)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:175)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.run(Phase.java:49)
	at jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:468)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecodeForStage(AnalysisParsedGraph.java:210)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:149)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.lambda$static$0(AnalysisMethod.java:1190)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.setGraph(AnalysisMethod.java:1273)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.createAnalysisParsedGraph(AnalysisMethod.java:1256)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1162)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.createAnalysisParsedGraph(AnalysisMethod.java:1252)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1162)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1023)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1015)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.decodeGraph(SimulateClassInitializerSupport.java:466)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.addClassInitializerDependencies(SimulateClassInitializerSupport.java:452)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:362)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:216)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.processClassInitializer(InlineBeforeAnalysisGraphDecoderImpl.java:98)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.handleEnsureClassInitializedNode(InlineBeforeAnalysisGraphDecoderImpl.java:74)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.doCanonicalizeFixedNode(InlineBeforeAnalysisGraphDecoderImpl.java:62)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.canonicalizeFixedNode(InlineBeforeAnalysisGraphDecoder.java:194)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.SimplifyingGraphDecoder.handleFixedNode(SimplifyingGraphDecoder.java:190)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:888)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysisGraphDecoder.java:269)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:609)
	at jdk.graal.compiler/jdk.graal.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:895)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:73)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:240)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:739)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:166)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:151)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraphInfo(MethodTypeFlow.java:109)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultAnalysisPolicy.staticRootMethodGraph(DefaultAnalysisPolicy.java:216)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.lambda$addRootMethod$1(PointsToAnalysis.java:392)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine$1.run(AbstractAnalysisEngine.java:338)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1750)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1742)
	at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1659)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool.externalHelpQuiesce(ForkJoinPool.java:2509)
	at java.base/java.util.concurrent.ForkJoinPool.helpQuiescePool(ForkJoinPool.java:2543)
	at java.base/java.util.concurrent.ForkJoinPool.awaitQuiescence(ForkJoinPool.java:4159)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.complete(CompletionExecutor.java:208)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.doTypeflow(PointsToAnalysis.java:682)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:648)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:160)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:841)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:569)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:534)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:548)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:761)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:163)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:111)
Caused by: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field io.netty.util.NetUtil.LOCALHOST4, accessors class io.netty.util.NetUtilSubstitutions$NetUtilLocalhost4Accessor: found no method named set or setLOCALHOST4
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:78)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.error(InjectedAccessorsPlugin.java:153)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleField(InjectedAccessorsPlugin.java:90)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleStoreStaticField(InjectedAccessorsPlugin.java:62)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5481)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5454)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5879)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3845)
	... 64 more
Internal exception: com.oracle.graal.pointsto.util.AnalysisError: parsing had failed in another thread
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisError.shouldNotReachHere(AnalysisError.java:189)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1174)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.createAnalysisParsedGraph(AnalysisMethod.java:1252)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1162)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1023)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1015)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.decodeGraph(SimulateClassInitializerSupport.java:466)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.addClassInitializerDependencies(SimulateClassInitializerSupport.java:452)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:362)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:216)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.analysis.DynamicHubInitializer.buildClassInitializationInfo(DynamicHubInitializer.java:245)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.analysis.DynamicHubInitializer.initializeMetaData(DynamicHubInitializer.java:127)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.analysis.NativeImagePointsToAnalysis.initializeMetaData(NativeImagePointsToAnalysis.java:179)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.initializeMetaData(AnalysisUniverse.java:762)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisType.lambda$new$1(AnalysisType.java:350)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisFuture.ensureDone(AnalysisFuture.java:69)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.analysis.NativeImagePointsToAnalysis.lambda$onTypeReachable$0(NativeImagePointsToAnalysis.java:142)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1750)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1742)
	at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1659)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: jdk.graal.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field io.netty.util.NetUtil.LOCALHOST4, accessors class io.netty.util.NetUtilSubstitutions$NetUtilLocalhost4Accessor: found no method named set or setLOCALHOST4
	at parsing io.netty.util.NetUtil.<clinit>(NetUtil.java:139)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2802)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:555)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3853)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.iterateBytecodesForBlock(SharedGraphBuilderPhase.java:1269)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3805)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3652)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.build(BytecodeParser.java:1193)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.build(SharedGraphBuilderPhase.java:245)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1076)
	at jdk.graal.compiler/jdk.graal.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:103)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:175)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.run(Phase.java:49)
	at jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:468)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.graal.compiler/jdk.graal.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecodeForStage(AnalysisParsedGraph.java:210)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:149)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.lambda$static$0(AnalysisMethod.java:1190)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.setGraph(AnalysisMethod.java:1273)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.createAnalysisParsedGraph(AnalysisMethod.java:1256)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1162)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.createAnalysisParsedGraph(AnalysisMethod.java:1252)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsedHelper(AnalysisMethod.java:1162)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1023)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:1015)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.decodeGraph(SimulateClassInitializerSupport.java:466)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.addClassInitializerDependencies(SimulateClassInitializerSupport.java:452)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:362)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.SimulateClassInitializerSupport.trySimulateClassInitializer(SimulateClassInitializerSupport.java:216)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.processClassInitializer(InlineBeforeAnalysisGraphDecoderImpl.java:98)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.handleEnsureClassInitializedNode(InlineBeforeAnalysisGraphDecoderImpl.java:74)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InlineBeforeAnalysisGraphDecoderImpl.doCanonicalizeFixedNode(InlineBeforeAnalysisGraphDecoderImpl.java:62)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.canonicalizeFixedNode(InlineBeforeAnalysisGraphDecoder.java:194)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.SimplifyingGraphDecoder.handleFixedNode(SimplifyingGraphDecoder.java:190)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:888)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysisGraphDecoder.java:269)
	at jdk.graal.compiler/jdk.graal.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:609)
	at jdk.graal.compiler/jdk.graal.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:895)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:73)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:240)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:739)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:166)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:151)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraphInfo(MethodTypeFlow.java:109)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultAnalysisPolicy.staticRootMethodGraph(DefaultAnalysisPolicy.java:216)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.lambda$addRootMethod$1(PointsToAnalysis.java:392)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine$1.run(AbstractAnalysisEngine.java:338)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:166)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:152)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1750)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.compute(ForkJoinTask.java:1742)
	at java.base/java.util.concurrent.ForkJoinTask$InterruptibleTask.exec(ForkJoinTask.java:1659)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
	at java.base/java.util.concurrent.ForkJoinPool.externalHelpQuiesce(ForkJoinPool.java:2509)
	at java.base/java.util.concurrent.ForkJoinPool.helpQuiescePool(ForkJoinPool.java:2543)
	at java.base/java.util.concurrent.ForkJoinPool.awaitQuiescence(ForkJoinPool.java:4159)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.complete(CompletionExecutor.java:208)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.doTypeflow(PointsToAnalysis.java:682)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:648)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:160)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:841)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:569)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:534)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:548)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:761)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:163)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:111)
Caused by: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field io.netty.util.NetUtil.LOCALHOST4, accessors class io.netty.util.NetUtilSubstitutions$NetUtilLocalhost4Accessor: found no method named set or setLOCALHOST4
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:78)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.error(InjectedAccessorsPlugin.java:153)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleField(InjectedAccessorsPlugin.java:90)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleStoreStaticField(InjectedAccessorsPlugin.java:62)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5481)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5454)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5879)
	at jdk.graal.compiler/jdk.graal.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3845)
	... 64 more
------------------------------------------------------------------------------------------------------------------------
                       0.6s (4.7% of total time) in 44 GCs | Peak RSS: 831.29MB | CPU load: 3.32
========================================================================================================================
Failed generating 'io.netty.netty-transport-native-unix-common_tests-tests' after 12.9s.

> Task :nativeTestCompile FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 5 executed, 10 up-to-date
com.oracle.svm.driver.NativeImage$NativeImageError
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.showError(NativeImage.java:2514)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.build(NativeImage.java:2089)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:2048)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.main(NativeImage.java:2030)
	at java.base@25.0.3/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTestCompile'.
> Process 'command '/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/bin/native-image'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 16s

> Task :nativeTestCompile FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTestCompile'.
> Native test compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 18s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image build' for 4.1.54.Final failed with exit code 1.
FAILED[native-image build][4.1.54.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.54.Final" ./gradlew clean nativeTestCompile -Pcoordinates="io.netty:netty-transport-native-unix-common:4.1.42.Final"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.57.Final" ./gradlew clean javaTest -Pcoordinates="io.netty:netty-transport-native-unix-common:4.1.42.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25239750106/job/74013375111
Last 300 lines of the log:
```console
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :jar
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

Netty_transport_native_unix_commonTest > fileDescriptorWrapsIntegerDescriptorWithValueSemanticsWithoutClosingIt() FAILED
    java.lang.UnsatisfiedLinkError: failed to load the required native library
        at io.netty.channel.unix.Unix.ensureAvailability(Unix.java:111)
        at io.netty.channel.unix.FileDescriptor.<init>(FileDescriptor.java:54)
        at io_netty.netty_transport_native_unix_common.Netty_transport_native_unix_commonTest.fileDescriptorWrapsIntegerDescriptorWithValueSemanticsWithoutClosingIt(Netty_transport_native_unix_commonTest.java:238)

        Caused by:
        java.lang.UnsatisfiedLinkError: could not load a native library: netty_transport_native_unix_linux_x86_64
            at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:226)
            at io.netty.channel.unix.Unix.loadNativeLibrary(Unix.java:84)
            at io.netty.channel.unix.Unix.<clinit>(Unix.java:56)
            ... 2 more

            Caused by:
            java.io.FileNotFoundException: META-INF/native/libnetty_transport_native_unix_linux_x86_64.so
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:175)
                ... 4 more

13 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/tests/test/index.html


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html
* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
13 actionable tasks: 3 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-transport-native-unix-common:4.1.42.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.57.Final failed with exit code 1.
SOFT-FAIL[native-image run][4.1.57.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.57.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-transport-native-unix-common:4.1.42.Final"]
========================================================================================
 io.netty:netty-transport-native-unix-common:4.1.42.Final:4.1.57.Final stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 731ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 945ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 io.netty:netty-transport-native-unix-common:4.1.42.Final:4.1.57.Final stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 754ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :testClasses

13 tests completed, 1 failed


> Task :test FAILED

Netty_transport_native_unix_commonTest > fileDescriptorWrapsIntegerDescriptorWithValueSemanticsWithoutClosingIt() FAILED
    java.lang.UnsatisfiedLinkError: failed to load the required native library
        at io.netty.channel.unix.Unix.ensureAvailability(Unix.java:111)
        at io.netty.channel.unix.FileDescriptor.<init>(FileDescriptor.java:54)
        at io_netty.netty_transport_native_unix_common.Netty_transport_native_unix_commonTest.fileDescriptorWrapsIntegerDescriptorWithValueSemanticsWithoutClosingIt(Netty_transport_native_unix_commonTest.java:238)

        Caused by:
        java.lang.UnsatisfiedLinkError: could not load a native library: netty_transport_native_unix_linux_x86_64
            at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:226)
            at io.netty.channel.unix.Unix.loadNativeLibrary(Unix.java:84)
            at io.netty.channel.unix.Unix.<clinit>(Unix.java:56)
            ... 2 more

            Caused by:
            java.io.FileNotFoundException: META-INF/native/libnetty_transport_native_unix_linux_x86_64.so
                at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:175)
                ... 4 more
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-transport-native-unix-common/4.1.42.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 4.1.57.Final failed with exit code 1.
FAILED[java run][4.1.57.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.57.Final" ./gradlew clean javaTest -Pcoordinates="io.netty:netty-transport-native-unix-common:4.1.42.Final"]
```

--- a/metadata/io.netty/netty-transport-native-unix-common/4.1.57.Final/reachability-metadata.json
+++ b/metadata/io.netty/netty-transport-native-unix-common/4.1.57.Final/reachability-metadata.json
@@ -1,0 +1,300 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.PreferredDirectByteBufAllocator"
+      },
+      "type": "io.netty.buffer.AbstractReferenceCountedByteBuf",
+      "fields": [
+        {
+          "name": "refCnt"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.FileDescriptor"
+      },
+      "type": "io.netty.channel.unix.FileDescriptor"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.Buffer",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.DatagramSocketAddress",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "byte[]",
+            "int",
+            "int",
+            "int",
+            "io.netty.channel.unix.DatagramSocketAddress"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.ErrorsStaticallyReferencedJniMethods",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.FileDescriptor",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.LimitsStaticallyReferencedJniMethods",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.unix.Socket",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "io.netty.channel.ChannelException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "java.lang.management.ManagementFactory",
+      "methods": [
+        {
+          "name": "getRuntimeMXBean",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "java.lang.management.RuntimeMXBean",
+      "methods": [
+        {
+          "name": "getInputArguments",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "java.nio.Bits"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.util.internal.PlatformDependent0$5"
+      },
+      "type": "java.nio.Bits",
+      "fields": [
+        {
+          "name": "UNALIGNED"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "java.nio.Buffer",
+      "fields": [
+        {
+          "name": "address"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.nio.Buffer",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "limit"
+        },
+        {
+          "name": "position"
+        }
+      ],
+      "methods": [
+        {
+          "name": "limit",
+          "parameterTypes": []
+        },
+        {
+          "name": "position",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "java.nio.DirectByteBuffer"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.nio.DirectByteBuffer",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "jdk.internal.misc.Unsafe",
+      "methods": [
+        {
+          "name": "getUnsafe",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "sun.management.VMManagementImpl",
+      "jniAccessible": true,
+      "fields": [
+        {
+          "name": "compTimeMonitoringSupport"
+        },
+        {
+          "name": "currentThreadCpuTimeSupport"
+        },
+        {
+          "name": "objectMonitorUsageSupport"
+        },
+        {
+          "name": "otherThreadCpuTimeSupport"
+        },
+        {
+          "name": "remoteDiagnosticCommandsSupport"
+        },
+        {
+          "name": "synchronizerUsageSupport"
+        },
+        {
+          "name": "threadAllocatedMemorySupport"
+        },
+        {
+          "name": "threadContentionMonitoringSupport"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "sun.misc.Unsafe",
+      "fields": [
+        {
+          "name": "theUnsafe"
+        }
+      ],
+      "methods": [
+        {
+          "name": "invokeCleaner",
+          "parameterTypes": [
+            "java.nio.ByteBuffer"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Buffer"
+      },
+      "type": "sun.misc.VM"
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.io.IOException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.lang.RuntimeException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.net.InetSocketAddress",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "java.lang.String",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.net.PortUnreachableException",
+      "jniAccessible": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.channel.unix.Unix"
+      },
+      "type": "java.nio.channels.ClosedChannelException",
+      "jniAccessible": true,
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "io.netty.util.internal.NativeLibraryLoader"
+      },
+      "glob": "META-INF/native/libnetty_transport_native_unix_linux_x86_64.so"
+    }
+  ]
+}

--- a/metadata/io.netty/netty-transport-native-unix-common/index.json
+++ b/metadata/io.netty/netty-transport-native-unix-common/index.json
@@ -1,6 +1,16 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "4.1.57.Final",
+    "tested-versions" : [
+      "4.1.57.Final"
+    ],
+    "allowed-packages" : [
+      "io.netty.channel.unix",
+      "io.netty.util.internal"
+    ]
+  },
+  {
     "metadata-version" : "4.1.42.Final",
     "source-code-url" : "https://repo.maven.apache.org/maven2/io/netty/netty-transport-native-unix-common/$version$/netty-transport-native-unix-common-$version$-sources.jar",
     "repository-url" : "https://github.com/netty/netty",

--- a/metadata/io.netty/netty-transport-native-unix-common/index.json
+++ b/metadata/io.netty/netty-transport-native-unix-common/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "4.1.57.Final",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/io/netty/netty-transport-native-unix-common/$version$/netty-transport-native-unix-common-$version$-sources.jar",
+    "repository-url" : "https://github.com/netty/netty",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/io/netty/netty-transport-native-unix-common-tests/$version$/netty-transport-native-unix-common-tests-$version$-sources.jar",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/io/netty/netty-transport-native-unix-common/$version$/netty-transport-native-unix-common-$version$-javadoc.jar",
+    "description" : "Netty Transport Native Unix Common provides shared JNI-backed Unix utility classes and native support used by Netty's Unix-based transport modules. It supplies common Unix socket address, file descriptor, and native library plumbing that higher-level Netty transports build on for asynchronous network clients and servers.",
     "tested-versions" : [
       "4.1.57.Final"
     ],

--- a/stats/io.netty/netty-transport-native-unix-common/4.1.57.Final/execution-metrics.json
+++ b/stats/io.netty/netty-transport-native-unix-common/4.1.57.Final/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-05-05": {
+    "timestamp": "2026-05-05T05:18:44.165774Z",
+    "library": "io.netty:netty-transport-native-unix-common:4.1.57.Final",
+    "starting_commit": "fedc689c0716754f5bb3f55458e1156efdc7292f",
+    "ending_commit": "bfc8257de0c3c0b83bcacedde44175f31e16feb5",
+    "previous_library": "io.netty:netty-transport-native-unix-common:4.1.42.Final",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "4.1.57.Final",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 497,
+          "missed": 1940,
+          "ratio": 0.203939,
+          "total": 2437
+        },
+        "line": {
+          "covered": 116,
+          "missed": 467,
+          "ratio": 0.198971,
+          "total": 583
+        },
+        "method": {
+          "covered": 45,
+          "missed": 126,
+          "ratio": 0.263158,
+          "total": 171
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "4.1.42.Final",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 425,
+          "missed": 1817,
+          "ratio": 0.189563,
+          "total": 2242
+        },
+        "line": {
+          "covered": 87,
+          "missed": 443,
+          "ratio": 0.164151,
+          "total": 530
+        },
+        "method": {
+          "covered": 40,
+          "missed": 123,
+          "ratio": 0.245399,
+          "total": 163
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 30035,
+      "cached_input_tokens_used": 296960,
+      "output_tokens_used": 2761,
+      "iterations": 1,
+      "cost_usd": 0.2313,
+      "tested_library_loc": 116,
+      "metadata_entries": 65,
+      "previous_library_metadata_entries": 29,
+      "code_coverage_percent": 19.9,
+      "previous_library_coverage_percent": 16.42
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/src/test/java/io_netty/netty_transport_native_unix_common/Netty_transport_native_unix_commonTest.java",
+      "metadata_file": "metadata/io.netty/netty-transport-native-unix-common/4.1.57.Final/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.netty/netty-transport-native-unix-common/4.1.57.Final/stats.json
+++ b/stats/io.netty/netty-transport-native-unix-common/4.1.57.Final/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "4.1.57.Final",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 497,
+        "missed" : 1940,
+        "ratio" : 0.203939,
+        "total" : 2437
+      },
+      "line" : {
+        "covered" : 116,
+        "missed" : 467,
+        "ratio" : 0.198971,
+        "total" : 583
+      },
+      "method" : {
+        "covered" : 45,
+        "missed" : 126,
+        "ratio" : 0.263158,
+        "total" : 171
+      }
+    }
+  } ]
+}

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/.gitignore
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/build.gradle
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.netty:netty-transport-native-unix-common:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/build.gradle
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/build.gradle
@@ -12,6 +12,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 
 dependencies {
     testImplementation "io.netty:netty-transport-native-unix-common:$libraryVersion"
+    testRuntimeOnly "io.netty:netty-transport-native-unix-common:$libraryVersion:linux-x86_64"
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/gradle.properties
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.netty:netty-transport-native-unix-common:4.1.57.Final
+library.version = 4.1.57.Final
+metadata.dir = io.netty/netty-transport-native-unix-common/4.1.57.Final/

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/settings.gradle
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.netty.netty-transport-native-unix-common_tests'

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/src/test/java/io_netty/netty_transport_native_unix_common/Netty_transport_native_unix_commonTest.java
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/src/test/java/io_netty/netty_transport_native_unix_common/Netty_transport_native_unix_commonTest.java
@@ -1,0 +1,257 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_netty.netty_transport_native_unix_common;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.unix.Buffer;
+import io.netty.channel.unix.DomainSocketAddress;
+import io.netty.channel.unix.DomainSocketReadMode;
+import io.netty.channel.unix.FileDescriptor;
+import io.netty.channel.unix.NativeInetAddress;
+import io.netty.channel.unix.PreferredDirectByteBufAllocator;
+import io.netty.channel.unix.UnixChannelOption;
+import io.netty.channel.unix.UnixChannelUtil;
+import java.io.File;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class Netty_transport_native_unix_commonTest {
+    @Test
+    void domainSocketAddressExposesPathAndValueSemantics() {
+        DomainSocketAddress stringAddress = new DomainSocketAddress("/tmp/netty-test.sock");
+        DomainSocketAddress equalAddress = new DomainSocketAddress(new File("/tmp/netty-test.sock"));
+        DomainSocketAddress otherAddress = new DomainSocketAddress("/tmp/netty-other.sock");
+
+        assertThat(stringAddress.path()).isEqualTo("/tmp/netty-test.sock");
+        assertThat(stringAddress.toString()).isEqualTo("/tmp/netty-test.sock");
+        assertThat(stringAddress).isEqualTo(equalAddress);
+        assertThat(stringAddress.hashCode()).isEqualTo(equalAddress.hashCode());
+        assertThat(stringAddress).isNotEqualTo(otherAddress);
+        assertThat(stringAddress).isNotEqualTo("/tmp/netty-test.sock");
+        assertThatNullPointerException()
+                .isThrownBy(() -> new DomainSocketAddress((String) null));
+    }
+
+    @Test
+    void nativeInetAddressMapsIpv4AddressToIpv4MappedIpv6Bytes() throws Exception {
+        InetAddress loopback = InetAddress.getByAddress(new byte[] {127, 0, 0, 1});
+        NativeInetAddress nativeAddress = NativeInetAddress.newInstance(loopback);
+        byte[] expected = new byte[] {
+                0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, (byte) 0xff, (byte) 0xff,
+                127, 0, 0, 1
+        };
+
+        assertThat(nativeAddress.address()).containsExactly(expected);
+        assertThat(nativeAddress.scopeId()).isZero();
+        assertThat(NativeInetAddress.ipv4MappedIpv6Address(loopback.getAddress())).containsExactly(expected);
+
+        byte[] copied = new byte[16];
+        NativeInetAddress.copyIpv4MappedIpv6Address(new byte[] {10, 20, 30, 40}, copied);
+        assertThat(copied).containsExactly(
+                0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, (byte) 0xff, (byte) 0xff,
+                10, 20, 30, 40);
+    }
+
+    @Test
+    void nativeInetAddressPreservesIpv6BytesAndScope() throws Exception {
+        byte[] ipv6Bytes = new byte[] {
+                0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 1
+        };
+        Inet6Address loopback = Inet6Address.getByAddress(null, ipv6Bytes, 0);
+        NativeInetAddress nativeAddress = NativeInetAddress.newInstance(loopback);
+
+        assertThat(nativeAddress.address()).containsExactly(ipv6Bytes);
+        assertThat(nativeAddress.scopeId()).isEqualTo(loopback.getScopeId());
+
+        NativeInetAddress explicitScope = new NativeInetAddress(ipv6Bytes, 7);
+        assertThat(explicitScope.address()).containsExactly(ipv6Bytes);
+        assertThat(explicitScope.scopeId()).isEqualTo(7);
+    }
+
+    @Test
+    void nativeInetAddressDecodesIpv4SocketAddressWithOffset() throws Exception {
+        byte[] encoded = new byte[12];
+        int offset = 2;
+        encoded[offset] = (byte) 192;
+        encoded[offset + 1] = (byte) 168;
+        encoded[offset + 2] = 1;
+        encoded[offset + 3] = 25;
+        encodeInt(encoded, offset + 4, 8080);
+
+        InetSocketAddress socketAddress = NativeInetAddress.address(encoded, offset, 8);
+
+        assertThat(socketAddress.getAddress().getAddress()).containsExactly((byte) 192, (byte) 168, 1, 25);
+        assertThat(socketAddress.getPort()).isEqualTo(8080);
+    }
+
+    @Test
+    void nativeInetAddressDecodesIpv6SocketAddressWithScopeAndOffset() throws Exception {
+        byte[] encoded = new byte[30];
+        int offset = 3;
+        byte[] ipv6Bytes = new byte[] {
+                0x20, 0x01, 0x0d, (byte) 0xb8,
+                0, 0, 0, 0,
+                0, 0, 0, 0,
+                0, 0, 0, 5
+        };
+        System.arraycopy(ipv6Bytes, 0, encoded, offset, ipv6Bytes.length);
+        encodeInt(encoded, offset + 16, 9);
+        encodeInt(encoded, offset + 20, 443);
+
+        InetSocketAddress socketAddress = NativeInetAddress.address(encoded, offset, 24);
+
+        assertThat(socketAddress.getAddress()).isInstanceOf(Inet6Address.class);
+        assertThat(socketAddress.getAddress().getAddress()).containsExactly(ipv6Bytes);
+        assertThat(((Inet6Address) socketAddress.getAddress()).getScopeId()).isEqualTo(9);
+        assertThat(socketAddress.getPort()).isEqualTo(443);
+    }
+
+    @Test
+    void bufferAllocatesDirectByteBufferUsingNativeByteOrder() {
+        ByteBuffer buffer = Buffer.allocateDirectWithNativeOrder(32);
+
+        assertThat(buffer.isDirect()).isTrue();
+        assertThat(buffer.capacity()).isEqualTo(32);
+        assertThat(buffer.order()).isEqualTo(ByteOrder.nativeOrder());
+    }
+
+    @Test
+    void preferredAllocatorDelegatesToDirectBuffersForGenericAndIoAllocations() {
+        PreferredDirectByteBufAllocator allocator = new PreferredDirectByteBufAllocator();
+        allocator.updateAllocator(UnpooledByteBufAllocator.DEFAULT);
+        ByteBuf buffer = allocator.buffer(16, 64);
+        ByteBuf ioBuffer = allocator.ioBuffer(8);
+        ByteBuf directBuffer = allocator.directBuffer(4);
+        ByteBuf heapBuffer = allocator.heapBuffer(4);
+        CompositeByteBuf compositeBuffer = allocator.compositeBuffer(2);
+        CompositeByteBuf compositeHeapBuffer = allocator.compositeHeapBuffer(2);
+        CompositeByteBuf compositeDirectBuffer = allocator.compositeDirectBuffer(2);
+        compositeBuffer.addComponent(true, UnpooledByteBufAllocator.DEFAULT.directBuffer(1).writeByte(1));
+        compositeHeapBuffer.addComponent(true, UnpooledByteBufAllocator.DEFAULT.heapBuffer(1).writeByte(2));
+        compositeDirectBuffer.addComponent(true, UnpooledByteBufAllocator.DEFAULT.directBuffer(1).writeByte(3));
+
+        try {
+            assertThat(buffer.isDirect()).isTrue();
+            assertThat(buffer.capacity()).isEqualTo(16);
+            assertThat(buffer.maxCapacity()).isEqualTo(64);
+            assertThat(ioBuffer.isDirect()).isTrue();
+            assertThat(directBuffer.isDirect()).isTrue();
+            assertThat(heapBuffer.isDirect()).isFalse();
+            assertThat(compositeBuffer.isDirect()).isTrue();
+            assertThat(compositeHeapBuffer.isDirect()).isFalse();
+            assertThat(compositeDirectBuffer.isDirect()).isTrue();
+            assertThat(allocator.isDirectBufferPooled())
+                    .isEqualTo(UnpooledByteBufAllocator.DEFAULT.isDirectBufferPooled());
+            assertThat(allocator.calculateNewCapacity(33, 128))
+                    .isEqualTo(UnpooledByteBufAllocator.DEFAULT.calculateNewCapacity(33, 128));
+        } finally {
+            buffer.release();
+            ioBuffer.release();
+            directBuffer.release();
+            heapBuffer.release();
+            compositeBuffer.release();
+            compositeHeapBuffer.release();
+            compositeDirectBuffer.release();
+        }
+    }
+
+    @Test
+    void nativeInetAddressConstructsFromEncodedBytesWithDefaultScope() {
+        byte[] addressBytes = new byte[] {
+                0x20, 0x01, 0x0d, (byte) 0xb8,
+                0, 0, 0, 0,
+                0, 0, 0, 0,
+                0, 0, 0, 9
+        };
+        NativeInetAddress nativeAddress = new NativeInetAddress(addressBytes);
+
+        assertThat(nativeAddress.address()).containsExactly(addressBytes);
+        assertThat(nativeAddress.scopeId()).isZero();
+    }
+
+    @Test
+    void nativeInetAddressRejectsUnsupportedSocketAddressPayloadLength() {
+        byte[] encoded = new byte[16];
+
+        assertThatThrownBy(() -> NativeInetAddress.address(encoded, 0, 12))
+                .isInstanceOf(Error.class);
+    }
+
+    @Test
+    void unixChannelUtilKeepsRequestedHostNameWhenOsAddressIsAvailable() throws Exception {
+        InetSocketAddress requested = InetSocketAddress.createUnresolved("requested.example", 1111);
+        InetSocketAddress operatingSystemAddress = new InetSocketAddress(
+                InetAddress.getByAddress(new byte[] {10, 11, 12, 13}), 2222);
+
+        InetSocketAddress computed = UnixChannelUtil.computeRemoteAddr(requested, operatingSystemAddress);
+
+        assertThat(computed.getHostString()).isEqualTo("requested.example");
+        assertThat(computed.getAddress().getAddress()).containsExactly(10, 11, 12, 13);
+        assertThat(computed.getPort()).isEqualTo(2222);
+        assertThat(UnixChannelUtil.computeRemoteAddr(requested, null)).isSameAs(requested);
+    }
+
+    @Test
+    void unixChannelOptionsExposeTypedConstantsAndValidation() {
+        ChannelOption<Boolean> reusePort = UnixChannelOption.SO_REUSEPORT;
+        ChannelOption<DomainSocketReadMode> readMode = UnixChannelOption.DOMAIN_SOCKET_READ_MODE;
+
+        assertThat(reusePort).isSameAs(ChannelOption.valueOf(UnixChannelOption.class, "SO_REUSEPORT"));
+        assertThat(readMode).isSameAs(ChannelOption.valueOf(UnixChannelOption.class, "DOMAIN_SOCKET_READ_MODE"));
+        assertThat(reusePort.name()).endsWith("SO_REUSEPORT");
+        assertThat(readMode.name()).endsWith("DOMAIN_SOCKET_READ_MODE");
+        reusePort.validate(Boolean.TRUE);
+        readMode.validate(DomainSocketReadMode.BYTES);
+        assertThatNullPointerException().isThrownBy(() -> readMode.validate(null));
+    }
+
+    @Test
+    void domainSocketReadModeSupportsStableEnumLookup() {
+        assertThat(DomainSocketReadMode.values()).containsExactly(
+                DomainSocketReadMode.BYTES,
+                DomainSocketReadMode.FILE_DESCRIPTORS);
+        assertThat(DomainSocketReadMode.valueOf("BYTES")).isSameAs(DomainSocketReadMode.BYTES);
+        assertThat(DomainSocketReadMode.valueOf("FILE_DESCRIPTORS"))
+                .isSameAs(DomainSocketReadMode.FILE_DESCRIPTORS);
+    }
+
+    @Test
+    void fileDescriptorWrapsIntegerDescriptorWithValueSemanticsWithoutClosingIt() {
+        FileDescriptor descriptor = new FileDescriptor(7);
+        FileDescriptor equalDescriptor = new FileDescriptor(7);
+        FileDescriptor otherDescriptor = new FileDescriptor(8);
+
+        assertThat(descriptor.intValue()).isEqualTo(7);
+        assertThat(descriptor.isOpen()).isTrue();
+        assertThat(descriptor).isEqualTo(equalDescriptor);
+        assertThat(descriptor.hashCode()).isEqualTo(equalDescriptor.hashCode());
+        assertThat(descriptor).isNotEqualTo(otherDescriptor);
+        assertThat(descriptor.toString()).isEqualTo("FileDescriptor{fd=7}");
+        assertThatThrownBy(() -> new FileDescriptor(-1)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    private static void encodeInt(byte[] target, int index, int value) {
+        target[index] = (byte) (value >>> 24);
+        target[index + 1] = (byte) (value >>> 16);
+        target[index + 2] = (byte) (value >>> 8);
+        target[index + 3] = (byte) value;
+    }
+}

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/user-code-filter.json
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.netty.channel.unix.**"
+    }
+  ]
+}

--- a/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/user-code-filter.json
+++ b/tests/src/io.netty/netty-transport-native-unix-common/4.1.57.Final/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "io.netty.channel.unix.**"
+    },
+    {
+      "includeClasses" : "io_netty.netty_transport_native_unix_common.**"
     }
   ]
 }

